### PR TITLE
Implement multithreading tests

### PR DESCRIPTION
**Pull request by MortimerGoro**
I have added some multithreading tests. It happened to me that context sharing was working ok in the current test (same thread) but was failing on servo due to multithreading environment. Better to test all the scenarios ;)

### CODE REVIEW

File: src/tests.rs [hunk 1]
```diff
@@ -158,6 +160,119 @@ fn test_sharing() {
...
+        gl::clear(gl::COLOR_BUFFER_BIT);
+
+        let vec = gl::read_pixels(0, 0, size.width, size.height, gl::RGBA, gl::UNSIGNED_BYTE);
+        test_pixels_eq(&vec, &[0, 255, 0, 255]);
+
+        sender.send(()).unwrap();
+
+        // Avoid drop until test ends
+        loop {
+            thread::sleep(time::Duration::from_millis(200));
```
Comment (emilio) [inline src/tests.rs]: This is nasty ;-)

Please add another kind of mechanism to block, another pair of channels would work.

**Comment by emilio** [comment]
Neat, looks good to me with that test change.

File: src/tests.rs (hunk 1 above)
Comment (MortimerGoro) [reply src/tests.rs]: Agree ;) Already fixed

**Comment by emilio** [comment]
r=me

**Pull request merged**